### PR TITLE
SWIFT-587 Retryable Reads

### PR DESCRIPTION
**Pull request by patrickfreed**
This PR exposes the `retryReads` option on `ClientOptions` to the public API and introduces the retryable reads test runner. It also updates much of our general spec test runner to work with crud v2 test files. As part of that, the retryable writes runner was separated into its own thing.

### CODE REVIEW

File: Tests/MongoSwiftTests/RetryableReadsTests.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        self.disableActiveFailPoint()
+    }
+
+    override func setUp() {
+        self.continueAfterFailure = false
+    }
+
+    func testRetryableReads() throws {
+        let skippedTestKeywords = [
+            "findOne",
```
Comment (kmahar) [inline Tests/MongoSwiftTests/RetryableReadsTests.swift]: I recently opened [SWIFT-643](https://jira.mongodb.org/browse/SWIFT-643) about adding a findOne method, add a TODO here?

Comment (patrickfreed) [reply Tests/MongoSwiftTests/RetryableReadsTests.swift]: done

File: Tests/MongoSwiftTests/RetryableReadsTests.swift [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            "changeStream", // TODO: SWIFT-648: Unskip this test
```
[the remark this answers is not in the document]
Comment (patrickfreed) [reply Tests/MongoSwiftTests/RetryableReadsTests.swift]: they run the test and apparently it passes. From talking with Kevin, it seems that maybe they're just not verifying all the results.

[the remark this answers is not in the document]
Comment (kmahar) [reply Tests/MongoSwiftTests/RetryableReadsTests.swift]: 🤷‍♀

**Pull request merged**